bundle-types/action.yml

```diff
@@ -0,0 +1,116 @@
+name: "Grafana Bundle Types"
+description: "Bundles a Grafana plugins types and pushes them to npm."
+
+outputs:
+  typesFilePath:
+    description: "The path to the built types file."
+    value: ${{ steps.set-outputs.outputs.typesFilePath }}
+
+inputs:
+  entryPoint:
+    description: "Location of types file to bundle."
+    required: false
+    default: "./src/types/index.ts"
+  tsConfig:
+    description: "A path to the tsconfig file to use when bundling types."
+    required: false
+    default: ""
+
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/setup-node@v4
+      with:
+        node-version: "${{ inputs.node-version }}"
+
+    - name: Install dependencies
+      run: ${{ github.action_path }}/pm.sh install
+      shell: bash
+
+    - name: Get plugin id and version
+      id: metadata
+      run: |
+        PLUGIN_ID=$(cat src/plugin.json | jq -r .id)
+        echo "plugin-id=${PLUGIN_ID}" >> $GITHUB_OUTPUT
+        PLUGIN_VERSION=$(cat package.json | jq -r .version)
+        echo "plugin-version=${PLUGIN_VERSION}" >> $GITHUB_OUTPUT
+      shell: bash
+
+    - id: get-secrets
```

Comment:
now that I read all the code what if we allow for third party to use this action as well?

my idea:

* Accept the github token as an input
* if we are inside a grafana org repository, run this step and overwrite its value with a generated one
* go on as usual

This way people outside of the grafana org who want to use this have the option to create a PAT and pass it and people in the grafana org simply use this without caring for the token parameter

Reply:
I realized we are opening a PR in a hardcode repo of us until now. Please ignore this suggestion.